### PR TITLE
Fixes the orm so it works with upgrades

### PR DESCRIPTION
**Pull request by WarlockD**
## About The Pull Request
Previous hotfix didn't let the orm work with upgrades and some of the other machinery didn't like it either.

## Why It's Good For The Game
We want an ungraded orm!
The previous was a hot fix, here is a more proper fix so please review.

## Changelog
:cl:
fix: Fixed the orm to work with upgrades
/:cl:

### CODE REVIEW

**Comment by kopoba** [comment]
Speedmerge please this is critical

File: code/game/machinery/recycler.dm [hunk 1]
```diff
@@ -139,21 +139,23 @@
...
-			return
-		materials.insert_item(I, multiplier = (amount_produced / 100))
-		materials.retrieve_all()
+		var/datum/component/material_container/mat_container = GetComponent(/datum/component/material_container)
+		var/material_amount = mat_container.get_item_material_amount(O)
+
+		if (material_amount)
+			var/loaded_amount = mat_container.insert_item(O, material_amount, multiplier = (amount_produced / 100))
+			if (loaded_amount >= material_amount) //if there is not enough space, eject it
+				mat_container.retrieve_all()
```
Comment (willox) [inline code/game/machinery/recycler.dm:157]: It looks like the old behaviour was to always empty the container after recycling something. Probably don't need the if-statement?

File: code/datums/components/material_container.dm [hunk 2]
```diff
@@ -116,7 +116,7 @@
 		user.put_in_active_hand(I)
 
 /// Proc specifically for inserting items, returns the amount of materials entered.
-/datum/component/material_container/proc/insert_item(obj/item/I, var/multiplier = 1, amt)
+/datum/component/material_container/proc/insert_item(obj/item/I, amt=1, multiplier = 1)
```
Comment (willox) [inline code/datums/components/material_container.dm:119]: This proc could use better documentation given how much trouble it gave us

**Review by sohpeach — COMMENTED** [review]
also idk about the unused `var/ore_multiplier` and `var/coin_multiplier` you added but they're not a big deal

**Comment by WarlockD** [comment]
Ugh I should of did another branch than add it to the exisiting one

**Comment by WarlockD** [comment]
> also idk about the unused `var/ore_multiplier` and `var/coin_multiplier` you added but they're not a big deal

Not sure they would ever be used but thought it be a good idea to put them in there so next time someone looks at it, they have an understanding of the function calls.

File: code/datums/components/material_container.dm [hunk 3]
```diff
@@ -115,7 +115,15 @@
 	else if(I == active_held)
 		user.put_in_active_hand(I)
 
-/// Proc specifically for inserting items, returns the amount of materials entered.
+/*
```
Comment (spookydonut) [inline code/datums/components/material_container.dm:118]: this isnt code docs needs to be /**

File: code/game/machinery/recycler.dm [hunk 4]
```diff
@@ -1,202 +1,201 @@
...
-	flags_1 = NODECONSTRUCT_1
-
-/obj/item/paper/guides/recycler
-	name = "paper - 'garbage duty instructions'"
-	info = "<h2>New Assignment</h2> You have been assigned to collect garbage from trash bins, located around the station. The crewmembers will put their trash into it and you will collect the said trash.<br><br>There is a recycling machine near your closet, inside maintenance; use it to recycle the trash for a small chance to get useful minerals. Then deliver these minerals to cargo or engineering. You are our last hope for a clean station, do not screw this up!"
-
-#undef SAFETY_COOLDOWN
+#define SAFETY_COOLDOWN 100
+
+/obj/machinery/recycler
```
Comment (spookydonut) [inline code/game/machinery/recycler.dm:3]: you have changed the line endings in this file

File: code/modules/mining/machine_processing.dm [hunk 5]
```diff
@@ -126,6 +126,7 @@
 	var/datum/material/selected_material = null
 	var/selected_alloy = null
 	var/datum/techweb/stored_research
+	var/ore_multiplier = 1				/// Just moved over from the orm
```
Comment (spookydonut) [inline code/modules/mining/machine_processing.dm:129]: do a proper doc comment

File: code/modules/mining/mint.dm [hunk 6]
```diff
@@ -14,7 +14,7 @@
 	var/produced_coins = 0 // how many coins the machine has made in it's last cycle
 	var/processing = FALSE
 	var/chosen = /datum/material/iron //which material will be used to make coins
-
+	var/coin_multiplier = 1			// not used
```
Comment (spookydonut) [inline code/modules/mining/mint.dm:17]: then remove it?

**Comment by LemonInTheDark** [comment]
Think this fixes #51344 ?

**Comment by WarlockD** [comment]
Damn didn't know about that one.  I made a hotfix pr to get the orm to work after #50951 was merged and I missed the fact that it doesn't work without upgrades.  Is that related to #50951?  I will look at it in a few

**Comment by WarlockD** [comment]
Well fuck it.  I fucked up the revert and now git has gone batshit.  As soon as my other outstanding pr's are merged, I will do a clean pr of this one from a proper branch.

**Pull request closed without merging**